ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/MethodModel.java

```diff
@@ -320,13 +329,13 @@ private WebTarget addMatrixParams(WebTarget webTarget, Object[] args) {
 
     private MultivaluedMap<String, Object> addCustomHeaders(Object[] args) {
         MultivaluedMap<String, Object> result = new MultivaluedHashMap<>();
-        for (Map.Entry<String, List<String>> entry : resolveCustomHeaders(args).entrySet()) {
-            entry.getValue().forEach(val -> result.add(entry.getKey(), val));
-        }
         for (String produce : produces) {
             result.add(HttpHeaders.ACCEPT, produce);
         }
-        result.add(HttpHeaders.CONTENT_TYPE, consumes[0]);
+        result.putSingle(HttpHeaders.CONTENT_TYPE, consumes[0]);
+        for (Map.Entry<String, List<String>> entry : resolveCustomHeaders(args).entrySet()) {
+            entry.getValue().forEach(val -> result.putSingle(entry.getKey(), val));
```

Comment:
Unless your intention is to only use the last header value, you should do:
`entry.getValue().forEach(val -> result.add(entry.getKey(), val));`